src/tests/unit/components/enhanced-table/enhanced-table-pagination/EnhancedTablePagination.spec.jsx

```diff
@@ -0,0 +1,46 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { renderHook } from '@testing-library/react'
+import { describe, it, expect } from 'vitest'
+
+import EnhancedTablePagination from '~/components/enhanced-table/enhanced-table-pagination/EnhancedTablePagination'
+import usePagination from '~/hooks/table/use-pagination'
+
+const itemsCount = 100
+
+describe('EnhancedTablePagination test', () => {
+  it('Should render first page', () => {
+    const { result } = renderHook(() => usePagination({ itemsCount }))
+
+    render(<EnhancedTablePagination pagination={result.current} />)
+
+    expect(screen.getByText(`1-5 table.of ${itemsCount}`)).toBeInTheDocument()
+  })
+
+  it('Should change page from 1 to 2', async () => {
+    const { result } = renderHook(() => usePagination({ itemsCount }))
+
+    const { rerender } = render(
+      <EnhancedTablePagination pagination={result.current} />
+    )
+    const inputField = screen.getByTestId('pagination-page-input')
+
+    fireEvent.change(inputField, { target: { value: 2 } })
+
+    rerender(<EnhancedTablePagination pagination={result.current} />)
```

Comment:
why this rerender is needed?

Reply:
Natalya, I’ve rewritten the test with userEvent, and now there's one more rerender. I can’t make it work without them, probably because this component isn't used in the app at all